### PR TITLE
[ci] use lintrunner in CI

### PR DESCRIPTION
**Pull request by suo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #68460

This changes our lint workflows to use lintrunner for the linters that
are currently supported

+ some random fixes to make things lint clean on master
+ changes to Makefile to use lintrunner

Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

### CODE REVIEW

**Comment by suo** [comment]
@suo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32470350).

**Review by t10-13rocket — APPROVED** [review]
LGTM

**Review by seemethere — APPROVED** [review]
w00t!

File: Makefile [hunk 1]
```diff
@@ -51,12 +52,7 @@ shellcheck:
 		--job 'shellcheck'
 
 setup_lint:
-	$(PYTHON) tools/actions_local_runner.py --file .github/workflows/lint.yml \
-		--job 'flake8-py3' --step 'Install dependencies' --no-quiet
-	$(PYTHON) tools/actions_local_runner.py --file .github/workflows/lint.yml \
-		--job 'cmakelint' --step 'Install dependencies' --no-quiet
-	$(PYTHON) tools/actions_local_runner.py --file .github/workflows/lint.yml \
-		--job 'mypy' --step 'Install dependencies' --no-quiet
+	$(PIP) install lintrunner
```
Comment (janeyx99) [inline Makefile:55]: soooo clean

**Review by janeyx99 — APPROVED** [review]

**Pull request closed without merging**